roles/jd-server/src/lib/mempool/mod.rs

```diff
@@ -43,31 +43,44 @@ impl JDsMempool {
     }
     pub fn new(url: String, username: String, password: String) -> Self {
         let auth = Auth::UserPass(username, password);
-        let empty_mempool: Vec<TransacrtionWithHash> = Vec::new();
+        let empty_mempool: HashMap<Txid, Option<Transaction>> = HashMap::new();
         JDsMempool {
             mempool: empty_mempool,
             auth,
             url,
         }
     }
 
+    pub fn add_tx_data_to_mempool(
+        self_: Arc<Mutex<Self>>,
+        txid: Txid,
+        transaction: Option<Transaction>,
+    ) {
+        let _ = self_.safe_lock(|x| {
+            x.mempool.insert(txid, transaction);
+        });
+    }
+
     pub async fn update_mempool(self_: Arc<Mutex<Self>>) -> Result<(), JdsMempoolError> {
-        let mut mempool_ordered: Vec<TransacrtionWithHash> = Vec::new();
+        let mut mempool_ordered: HashMap<Txid, Option<Transaction>> = HashMap::new();
         let client = self_
             .safe_lock(|x| x.get_client())
             .unwrap()
             .ok_or(JdsMempoolError::NoClient)?;
-        let new_mempool: Result<Vec<TransacrtionWithHash>, JdsMempoolError> =
+        let self_clone = self_.clone();
+        let new_mempool: Result<HashMap<Txid, Option<Transaction>>, JdsMempoolError> =
```

Comment:
you can do:
```rust
let new_mempool: Result<HashMap<Txid, Option<Transaction>>, JdsMempoolError> = {
    let self_ = self_.clone();
    ...
}
 ```
 instead of `let self_clone = self_.clone();`

Reply:
Done 4be66e0e70c7f392b70fbb0f3df8eb18dcc4fed8